File: code_examples/5-Holding_Data/1-Arrays/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses an array
 */

package main

import (
	"fmt"
)

func main() {
	// this function uses an array

	var studentMarks [5]int
	var aSingleMark int

	arrayLength := len(studentMarks)

	// input
	for loopCounter := 0; loopCounter < arrayLength; loopCounter++ {
		fmt.Print("Enter a mark (percentage): ")
		fmt.Scan(&aSingleMark)
		studentMarks[loopCounter] = aSingleMark
	}

	fmt.Println()
	fmt.Println("Here are the 5 marks:")

	// get length of array
	for loopCounter := 0; loopCounter < arrayLength; loopCounter++ {
		fmt.Printf("%d%% ", studentMarks[loopCounter])
	}

	fmt.Println()
	fmt.Println("\nDone.")
}

File: code_examples/5-Holding_Data/1-Arrays/CSharp/main.cs
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses an array

using System;

class Program
{
    static void Main() {
        // this function uses an array

        int[] studentMarks = new int[5];
        int aSingleMark;

        // get length of array
        int arrayLength = studentMarks.Length;

        // input
        for (int loop_counter = 0; loop_counter < arrayLength; loop_counter++) {
            Console.Write("Enter a mark (percentage): ");
            aSingleMark = Convert.ToInt32(Console.ReadLine());
            studentMarks[loop_counter] = aSingleMark;
        }

        Console.WriteLine();
        Console.WriteLine("Here are the 5 marks:");

        for (int loop_counter = 0; loop_counter < arrayLength; loop_counter++) {
            Console.Write(studentMarks[loop_counter] + "% ");
        }

        Console.WriteLine();
        Console.WriteLine("\nDone.");
    }
}
